src/main/java/frc/lib/lib2706/ProfiledPIDFFController.java

```diff
@@ -0,0 +1,57 @@
+package frc.lib.lib2706;
+
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.controller.SimpleMotorFeedforward;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.wpilibj.Timer;
+import frc.robot.Config;
+import frc.robot.Config.ArmConfig;
+
+public class ProfiledPIDFFController {
+    private double lastVel = 0; 
+    private double lastTime = Timer.getFPGATimestamp();
+
+    private final TrapezoidProfile.Constraints m_constraints = 
+    new TrapezoidProfile.Constraints(Config.ArmConfig.MAX_VEL, Config.ArmConfig.MAX_ACCEL);
+    private final ProfiledPIDController m_ProfiledPIDController = 
+    new ProfiledPIDController(1.6,0.002,40, m_constraints, 0.02);
+
+    private final SimpleMotorFeedforward m_FFController = new SimpleMotorFeedforward(0,0);
+
+    public ProfiledPIDFFController ()
+    {
+
+    }
+
+    public void setFinalGoal( TrapezoidProfile.State finalGoal) 
+    {
+        m_ProfiledPIDController.setGoal(finalGoal);
+    }
+
+    public double getProfiledPIDVoltage (double measuredPos, double finalPos)
+    {
+        return m_ProfiledPIDController.calculate(measuredPos, finalPos);
+    }
+
+    public double getProfiledPIDFFVoltage(double measuredPos, double finalPos)
+    {
+        double pidVal = getProfiledPIDVoltage(measuredPos, finalPos);
+
+        double currVel = m_ProfiledPIDController.getSetpoint().velocity;
+        double currAccel = (m_ProfiledPIDController.getSetpoint().velocity - lastVel)/(Timer.getFPGATimestamp() - lastTime);
```

Comment:
I found that using a measured time delta causes more headaches than it's worth. Sometimes its not used in awhile so it doesn't update the last time and then the first time you use it, it has a large time delta.

It's easier to just divide by 0.02 instead.

Reply:
Ideally, it's even better to use the other version of SimpleMotorFeedforward that takes the previous velocity and current velocity. https://github.com/wpilibsuite/allwpilib/blob/ee15cc172a5ee8e7e8099b7e8d5fe7d3fadcc828/wpimath/src/main/java/edu/wpi/first/math/controller/SimpleMotorFeedforward.java#L74